src/prng.c

```diff
@@ -0,0 +1,99 @@
+/******************************************************************************
+ *
+ * Copyright 2017 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <ecdaa/prng.h>
+
+#ifndef DISABLE_LIBSODIUM_RNG_SEED_FUNCTION
+
+#include <sodium.h>
+
+#if defined(__linux__)
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <linux/random.h>
+#endif  // defined(__linux__)
+
+static int check_entropy();
+
+int ecdaa_prng_init(struct ecdaa_prng *prng_in)
+{
+    int ret = 0;
+
+    prng_in->initialized = ECDAA_PRNG_INITIALIZED_NO;
+
+    do {
+        ret = check_entropy();
+        if (0 != ret)
+            break;
```

Comment:
I'd personally use `goto`s rather than ` do/while/break` to avoid the extra level of indentation, but that's a personal preference.

```
ret = check_entropy();
if (0 != ret)
  goto out;

...

prng_in->initialized = ECDAA_PRNG_INITIALIZED_YES;

out:
  return ret;
```

Reply:
The `do-while` construct appears in other crypto software I've seen, that's why I adopted it. I kind of like the extra indentation, because it clearly shows what's wrapped (like a try-catch block).

Reply:
👍